log4j-api-kotlin/src/main/kotlin/org/apache/logging/log4j/kotlin/LoggingFactory.kt

```diff
@@ -75,3 +81,5 @@ private fun <T : Any> unwrapCompanionClass(ofClass: Class<T>): Class<*> {
     ofClass
   }
 }
+
+private val loggerCache = ConcurrentHashMap<Class<*>, KotlinLogger>()
```

Comment:
The easiest implementation of an LRU cache in Java would be using https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html#removeEldestEntry-java.util.Map.Entry- with `Collections.synchronizedMap()`. For a more sophisticated cache, copying some code from commons-collections could be done.

Reply:
Thanks for the link -- I've never actually come across this approach before! Doing it unsophisticated for now is probably fine -- I've committed 7783beb. Note this is a bit slower than `ConcurrentHashMap` -- the benchmarks show 14.1 ns/op vs 5 ns/op, but I think its ok. Still almost an order of magnitude better than now!